### PR TITLE
rewrite to go

### PR DESCRIPTION
**Pull request by mkyc**

### CODE REVIEW

**Review by seriva — DISMISSED** [review]

File: cmd/init.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		fmt.Println("Initialized config: \n" + string(bytes))
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(initCmd)
+
+	initCmd.Flags().String("name", "epiphany", "prefix given to all resources created") //TODO rename to prefix
+	initCmd.Flags().String("region", "eu-central-1", "region used to initialize config file")
+	initCmd.Flags().String("vms_rsa", "vms_rsa", "name of rsa keypair to be provided to machines")
```
Comment (atsikham) [inline cmd/init.go:111]: I would choose --vms-rsa, but up to you

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        github.com/epiphany-platform/e-structures => ../../epiphany-platform/e-structures
+    )
+    ```
+# Develop terraform scripts
+
+There is a simple way to develop terraform scripts independently of module.
+
+1) have image built already
+1) go to `resources` directory in terminal
+1) run module init command ie.: `docker run --rm -v $(pwd)/shared:/shared -t epiphanyplatform/awsbi:dev init`
```
Comment (atsikham) [inline docs/DEVELOPMENT.md:120]: i.e.

File: resources/terraform/modules/vm_group/output.tf [hunk 3]
```diff
@@ -6,7 +6,8 @@ output "vm_group" {
     {
       vm_name: vm.tags.Name
       public_ip: vm.public_ip
-      private_ips: vm.private_ip
+      private_ip: vm.private_ip
```
Comment (atsikham) [inline resources/terraform/modules/vm_group/output.tf:9]: So don't we support a few private IPs for 1 VM?

**Review by atsikham — CHANGES_REQUESTED** [review]
Shouldn't examples/basic_flow be removed? Or it will be done later in the scope of other work?

File: resources/terraform/modules/vm_group/output.tf (hunk 3 above)
Comment (mkyc) [reply resources/terraform/modules/vm_group/output.tf:9]: code didn't support anyways. I just changed name of variable to reflect value retrieved from vm. I also checked terraform resource and it is clear that it has one IP. We probably should extract it to separate issue.

File: Dockerfile [hunk 4]
```diff
@@ -1,29 +1,38 @@
+# golang builder
+FROM golang:1.15.2 as builder
```
[the remark this answers is not in the document]
Comment (mkyc) [reply Dockerfile:2]: I don't want to update all versions here. This PR is big enough. I will exclude it to separate issue.

File: Dockerfile [hunk 5]
```diff
@@ -1,29 +1,38 @@
...
+ENV M_VERSION=$ARG_M_VERSION
+RUN mkdir -p $GOPATH/src/$GO_MODULE_NAME
+COPY . $GOPATH/src/$GO_MODULE_NAME
+WORKDIR $GOPATH/src/$GO_MODULE_NAME
+RUN go get -v
+RUN go get github.com/ahmetb/govvv
+RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w $(govvv -flags -pkg $GO_MODULE_NAME/cmd -version $M_VERSION)" -x -o /m-aws-basic-infrastructure $GO_MODULE_NAME
+
+# terraform init
 FROM hashicorp/terraform:0.13.2 as initializer
```
[the remark this answers is not in the document]
Comment (mkyc) [reply Dockerfile:15]: same as above

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply Dockerfile:2]: added #67

File: Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply Dockerfile:15]: also referenced in #67

File: resources/terraform/modules/vm_group/output.tf (hunk 3 above)
Comment (mkyc) [reply resources/terraform/modules/vm_group/output.tf:9]: created #68

**Review by atsikham — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Pull request merged**